src/main/xsl/preprocess/flagImpl.xsl

```diff
@@ -87,51 +93,50 @@ LOOK FOR FIXME TO FIX SCHEMEDEF STUFF
     </xsl:param>
     <xsl:choose>
       <xsl:when test="exists($flagrules/*)">
-        <xsl:variable name="conflictexist">
+        <xsl:variable name="conflictexist" as="xs:boolean">
          <xsl:call-template name="conflict-check">
-          <xsl:with-param name="flagrules" select="$flagrules"/>
+           <xsl:with-param name="flagrules" as="element()*" select="$flagrules"/>
          </xsl:call-template>
         </xsl:variable>
         <xsl:copy>
           <xsl:apply-templates select="@*"/>
           <ditaval-startprop class="+ topic/foreign ditaot-d/ditaval-startprop ">
             <xsl:apply-templates select="." mode="gen-style">
-              <xsl:with-param name="flagrules" select="$flagrules"/>
+              <xsl:with-param name="flagrules" as="element()*" select="$flagrules"/>
               <xsl:with-param name="conflictexist" select="$conflictexist"/>
             </xsl:apply-templates>
-            <xsl:if test="$conflictexist='true' and $FILTERDOC/val/style-conflict">
-              <xsl:copy-of select="$FILTERDOC/val/style-conflict"/>
+            <xsl:if test="$conflictexist and $FILTERDOC/val/style-conflict">
+              <xsl:sequence select="$FILTERDOC/val/style-conflict"/>
             </xsl:if>
             <xsl:apply-templates select="." mode="dita-start-flagit">
-              <xsl:with-param name="flagrules" select="$flagrules"/>
+              <xsl:with-param name="flagrules"  as="element()*" select="$flagrules"/>
             </xsl:apply-templates>
             <xsl:apply-templates select="." mode="dita-start-revflag">
-              <xsl:with-param name="flagrules" select="$flagrules"/>
+              <xsl:with-param name="flagrules"  as="element()*" select="$flagrules"/>
             </xsl:apply-templates>
           </ditaval-startprop>
-          <xsl:apply-templates select="*|processing-instruction()|comment()|text()"/>
+          <xsl:apply-templates select="node()"/>
           <ditaval-endprop class="+ topic/foreign ditaot-d/ditaval-endprop ">
             <xsl:apply-templates select="." mode="dita-end-revflag">
-              <xsl:with-param name="flagrules" select="$flagrules"/>
+              <xsl:with-param name="flagrules" as="element()*" select="$flagrules"/>
             </xsl:apply-templates>
             <xsl:apply-templates select="." mode="dita-end-flagit">
-              <xsl:with-param name="flagrules" select="$flagrules"/>
+              <xsl:with-param name="flagrules" as="element()*" select="$flagrules"/>
             </xsl:apply-templates>
           </ditaval-endprop>
         </xsl:copy>
       </xsl:when>
       <xsl:otherwise>
         <xsl:copy>
-          <xsl:apply-templates select="@*|*|processing-instruction()|comment()|text()"/>
+          <xsl:apply-templates select="@*,node()"/>
```

Comment:
Prefer `@* | node()` to `@*, node()`, since that's been the convention in the other stylesheets too.